src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -897,7 +898,7 @@ fn preprocess_link(
     // Parse and strip the disambiguator from the link, if present.
     let (disambiguator, path_str, link_text) = match Disambiguator::from_str(link) {
         Ok(Some((d, path, link_text))) => (Some(d), path.trim(), link_text.trim()),
-        Ok(None) => (None, link.trim(), link.trim()),
+        Ok(None) => (None, link, link),
```

Comment:
This is a behavior change, and also an inconsistency with the `compiler/rustc_resolve/src/rustdoc.rs` change.

I'd prefer to decide on what grammar we actually want here (and maybe document it) before making any changes.

Reply:
This is not. The `trim()` method is called just above.

Reply:
It makes a difference whether trim is called before `Disambiguator::from_str` or after.

Reply:
Well, it's needed for the fix. Where you want the documentation about this to put?

Reply:
It's not needed for the fix, all the test cases worked before.

I don't want to argue about this too much, it's a reasonable change to support `[ type@Path]`, could you just bring `compiler/rustc_resolve/src/rustdoc.rs` into consistency and do the trim in the same place, between `#` and `@`.

Reply:
Done!